Blog post on agentic AI with Quarkus - part 2

mariofusco opened the pull request:
@cescoffier @geoand Please review.

geoand commented on _posts/2025-02-20-agentic-ai-with-quarkus-p2.adoc:
```diff
@@ -0,0 +1,312 @@
+
+[source, java]
+----
+@ApplicationScoped
+public class BookingService {
+
+	private final int capacity;
+
+	public BookingService(@ConfigProperty(name = "restaurant.capacity") int capacity) {
+    	this.capacity = capacity;
+	}
+
+	public boolean hasCapacity(LocalDate date, int partySize) {
+    	int sum = Booking.find("date", date).list().stream().map(b -> (Booking) b)
+            	.mapToInt(b -> b.partySize)
+            	.sum();
+    	return sum + partySize <= capacity;
+	}
+
+	@Transactional
+	@Tool("Books a table for a given name, date (passed as day fo the month, month and year), party size and preference (indoor/outdoor). If the restaurant is full, an exception is thrown. If preference is not specified, `UNSET` is used.")
+	public String book(String name, int day, int month, int year, int partySize, Booking.Preference preference) {
+    	var date = LocalDate.of(year, month, day);
+    	if (hasCapacity(date, partySize)) {
+        	Booking booking = new Booking();
+        	booking.date = date;
+        	booking.name = name;
+        	booking.partySize = partySize;
+        	if (preference == null) {
+            	preference = Booking.Preference.UNSET;
+        	}
+        	booking.preference = preference;
+        	booking.persist();
+        	String result = String.format("%s successfully booked a %s table for %d persons on %s", name, preference, partySize, date);
+        	Log.info(result);
+        	return result;
+    	}
+    	return "The restaurant is full for that day";
+	}
```
This seems to have some formatting issues

donbourne commented on _posts/2025-02-20-agentic-ai-with-quarkus-p2.adoc:
```diff
@@ -0,0 +1,312 @@
+}
+----
+
+2 .  a https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/aiastool/geo/GeoCodingService.java[web service] returning the geographic coordinates of a given city.
```
Can you add a comment about how the use of a web service as a tool is significant? I can call the code as a web service, but when it's used as a Tool is it just treated the same way as any other Tool (where the agent framework calls the target method of the Tool directly)?

donbourne commented on _posts/2025-02-20-agentic-ai-with-quarkus-p2.adoc:
```diff
@@ -0,0 +1,312 @@
+
+=== The weather forecast agent
+
+This https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/aiastool[first example] of agentic AI implements the following https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/aiastool/WeatherForecastAgent.java[weather forecast agent]
```
would be good to introduce this section with a sentence about what your weather forecast agent is expected to accomplish.

donbourne commented on _posts/2025-02-20-agentic-ai-with-quarkus-p2.adoc:
```diff
@@ -0,0 +1,312 @@
+
+=== A more general purpose AI agent
+
+An AI agent can have even more freedom to operate, thus being more flexible and covering a wider range of tasks. Typically, these scenarios requires a bigger model, so https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/searchastool[this second example] has been configured with `qwen2.5-14b` and a longer timeout to give it a chance to complete its task.
```
would be good to introduce this section with a sentence describing the scenario that you're about to provide the technical details for.

donbourne commented on _posts/2025-02-20-agentic-ai-with-quarkus-p2.adoc:
```diff
@@ -0,0 +1,312 @@
+
+	@Tool("Perform a web search to retrieve information online")
+	String webSearch(String q) throws IOException {
+    	String webUrl = "https://html.duckduckgo.com/html/?q=" + q;
+    	return Jsoup.connect(webUrl).get().text();
```
indentation

mariofusco commented:
@cescoffier @geoand I reworked this a bit and also added the tracing for the weather example. I believe this is almost ready to be published, except for the conclusions that probably could be detailed a bit more? Please review it and if you have any suggestion let me know.

cescoffier commented:
I'm working on it

cescoffier commented:
I've rewrote some parts to try to better explain "agents". I also changed some pictures. 

@mariofusco let me know. 

I'm going to start a new pass focusing on the text (spellcheck, some minor rewrite)

cescoffier commented:
@mariofusco Done. Can you have a look to the preview (when updated)?

mariofusco commented:
@cescoffier Thank you, it was indeed missing some context around the concept of "agents" and this clarifies a lot. And I love the new images :) 

I didn't know know at all about Tavily, I'll give it a try in my example. This will be nice especially since I want to show that and other examples in my agentic AI that I will give at VoxxedDays Zurich and hopefully somewhere else. 

The content is ok for me. Let me know if you're going to add any other changes, and when you're happy with it I will squash all commits so it will be ready to be published.

cescoffier commented:
I did a last pass - nothing more. So, ready to be squashed and pushed and merged!

mariofusco commented:
Squashed and ready to be merged and published.
